tools/default.nix

```diff
@@ -0,0 +1,65 @@
+{ pkgs ? import <nixpkgs> {} , withEditor ? false }:
+
+pkgs.stdenv.mkDerivation rec {
+  name = "openclonk";
+
+  gitRef = pkgs.lib.commitIdFromGitRepo ../.git;
+
+  src = builtins.filterSource (path: type: ! builtins.elem (baseNameOf path) [
+    ".git" # leave out .git as it changes often in ways that do not affect the build
+    "default.nix" # default.nix might change, but the only thing that matters is what it evaluates to, and nix takes care of that
+    "result" # build result is irrelevant
+    "build"
+  ]) ./..;
+
+  enableParallelBuilding = true;
+
+  hardeningDisable = "format";
+
+  nativeBuildInputs = with pkgs; [ cmake ];
+
+  dontStrip = true;
+
+  buildInputs = with pkgs; [
+    SDL2 libvorbis libogg libjpeg libpng freetype glew tinyxml
+    openal freealut pkgconfig
+  ] ++ stdenv.lib.optional withEditor qt5.full;
+
+  cmakeFlags = [ "-DCMAKE_AR=${pkgs.gcc-unwrapped}/bin/gcc-ar" "-DCMAKE_RANLIB=${pkgs.gcc-unwrapped}/bin/gcc-ranlib" ];
+
+  preConfigure = ''
+    sed s/REVGOESHERE/''${gitRef:0:12}/ > cmake/GitGetChangesetID.cmake <<EOF
```

Comment:
For C4REVISION_DIRTY, I don't see a convenient way to extract that information using nixpkgs's library and I don't think it's feasible without including the git repo in the source — at which point we could just use the normal GitGetChangesetID — but at the cost of many more unnecessary rebuilds. For C4REVISION_TS, it would be easier to get at it with just nix, but it's not implemented anywhere as far as I can see and requires decompressing the zlib-compressed commit file, which seems like more effort than it's worth.